test/ergw_http_api_SUITE.erl

```diff
@@ -134,47 +203,265 @@
+	  ]}
+	]).
+-define(CONFIG_UPDATE,
+	[{[sockets, cp, ip], localhost},
+	 {[sockets, irx, ip], test_gsn},
+	 {[sockets, sx, ip], localhost},
+	 {[node_selection, {default, 2}, 2, "topon.gn.ggsn.$ORIGIN"],
+	  {fun node_sel_update/2, final_gsn}},
+	 {[node_selection, {default, 2}, 2, "topon.sx.prox01.$ORIGIN"],
+	  {fun node_sel_update/2, pgw_u01_sx}},
+	 {[node_selection, {default, 2}, 2, "topon.sx.prox02.$ORIGIN"],
+	  {fun node_sel_update/2, sgw_u_sx}}
 	]).
 
+node_sel_update(Node, {_,_,_,_} = IP) ->
+    {Node, [IP], []};
+node_sel_update(Node, {_,_,_,_,_,_,_,_} = IP) ->
+    {Node, [], [IP]}.
+
 all() ->
     [http_api_version_req,
      http_api_status_req,
      http_api_status_accept_new_get_req,
      http_api_status_accept_new_post_req,
-     http_api_prometheus_metrics_req
-     %% http_api_delete_sessions
+     http_api_prometheus_metrics_req,
+     http_api_delete_sessions,
+	 http_api_delete_all_sessions
     ].
 
-init_per_testcase(Config) ->
-    meck_reset(Config).
-init_per_testcase(http_api_delete_sessions, Config) ->
+%%%===================================================================
+%%% Tests
+%%%===================================================================
+
+setup_per_testcase(Config) ->
+    setup_per_testcase(Config, true).
+
+setup_per_testcase(Config, ClearSxHist) ->
     ct:pal("Sockets: ~p", [ergw_socket_reg:all()]),
     ergw_test_sx_up:reset('pgw-u01'),
     meck_reset(Config),
     start_gtpc_server(Config),
+    reconnect_all_sx_nodes(),
+    ClearSxHist andalso ergw_test_sx_up:history('pgw-u01', true),
+    ok.
+
+end_per_testcase(Config) ->
+    stop_gtpc_server(),
+
+    PoolId = [<<"pool-A">>, ipv4, "10.180.0.1"],
+    ?match_metric(prometheus_gauge, ergw_local_pool_free, PoolId, 65534),
+
+    AppsCfg = proplists:get_value(aaa_cfg, Config),
+    ok = application:set_env(ergw_aaa, apps, AppsCfg),
+    set_online_charging(false),
+    ok.
+
+set_online_charging([], true, Cfg)
+  when is_map(Cfg) ->
+    maps:put('Online', [1], Cfg);
+set_online_charging([], _, Cfg)
+  when is_map(Cfg) ->
+    maps:remove('Online', Cfg);
+set_online_charging([], _, Cfg) ->
+    Cfg;
+
+set_online_charging([Key|Next], Set, [{_, _}|_] = Cfg)
+  when Key =:= '_' ->
+    lists:map(
+      fun({K, V}) -> {K, set_online_charging(Next, Set, V)} end, Cfg);
+set_online_charging([Key|Next], Set, [{_, _}|_] = Cfg) ->
+    New = {Key, set_online_charging(Next, Set, proplists:get_value(Key, Cfg))},
+    lists:keystore(Key, 1, Cfg, New);
+
+set_online_charging([Key|Next], Set, Cfg)
+  when Key =:= '_', is_map(Cfg) ->
+    maps:map(
+      fun(_, V) -> set_online_charging(Next, Set, V) end, Cfg);
+set_online_charging([Key|Next], Set, Cfg)
+  when is_map(Cfg) ->
+    Cfg#{Key => set_online_charging(Next, Set, maps:get(Key, Cfg))}.
+
+set_online_charging(Set) ->
+    {ok, Cfg0} = application:get_env(ergw, charging),
+    Cfg = set_online_charging(['_', rulebase, '_'], Set, Cfg0),
+    ok = application:set_env(ergw, charging, Cfg).
+
```

Comment:
you are adding far too much ballast to to this suite. All this charging stuff and a lot of other things are not needed here.